File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        long long ans=0;
        long long n=nums.size();
        unordered_map<int,long long>mp;
        for(int i=0;i<n;i++){
            if(mp.find(nums[i]-i)!=mp.end()){
                ans+=mp[nums[i]-i];
            }
            mp[nums[i]-i]++;

        }
        long long tmp=n*(n-1)/2;
        return tmp-ans;
    }
};

File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.py
class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
        n=len(nums)
        mp={}
        ans=0
        tmp=n*(n-1)//2
        for i in range(n):
            x=i-nums[i]
            if x in mp:
                ans+=mp[x]
            mp[x]= mp.get(x,0)+1


        return tmp-ans
            

        